src/lib.rs

```diff
@@ -211,6 +211,47 @@ impl SdpSession {
         self.media.extend(v)
     }
 
+    pub fn parse_session_vector(&mut self, lines: &mut Vec<SdpLine>) -> Result<(), SdpParserError> {
+        while lines.len() > 0 {
+            let line = lines.remove(0);
+            match line.sdp_type {
+                SdpType::Attribute(ref a) => {
+                    self
+                        .add_attribute(a)
+                        .map_err(|e: SdpParserInternalError| SdpParserError::Sequence {
+                            message: format!("{}", e),
+                            line_number: line.line_number,
+                        })?
+                }
+                SdpType::Bandwidth(b) => self.add_bandwidth(b),
+                SdpType::Timing(t) => self.set_timing(t),
+                SdpType::Connection(c) => self.set_connection(c),
+
+                SdpType::Origin(_) | SdpType::Session(_) | SdpType::Version(_) => {
+                    return Err(SdpParserError::Sequence {
+                        message: "version, origin or session at wrong level".to_string(),
+                        line_number: line.line_number,
+                    });
+                }
+                SdpType::Media(_) => {
+                    return Err(SdpParserError::Sequence {
+                        message: "media line not allowed in session parser".to_string(),
+                        line_number: line.line_number,
+                    });
+                }
+                // the line parsers throw unsupported errors for these already
+                SdpType::Email(_)
```

Comment:
If these SDP types throw parsing errors, then they should be removed from the enum, and that way we don't have to worry about illegal state everywhere we use this enum. This follows the '"make the impossible inexpressible" rule of thumb. That could be handled in another bug though.

Reply:
Excellent idea. I created issue #107 for that!